exec/rpc/src/main/java/org/apache/drill/exec/rpc/RpcBus.java

```diff
@@ -159,19 +159,15 @@ public ChannelClosedHandler(C clientConnection, Channel channel) {
     @Override
     public void operationComplete(ChannelFuture future) throws Exception {
       String msg;
-      if(local!=null) {
+      if(local != null) {
         msg = String.format("Channel closed %s <--> %s.", local, remote);
       }else{
         msg = String.format("Channel closed %s <--> %s.", future.channel().localAddress(), future.channel().remoteAddress());
       }
 
-      if (RpcBus.this.isClient()) {
-        if(local != null) {
-          logger.info(String.format(msg));
-        }
-      } else {
-        queue.channelClosed(new ChannelClosedException(msg));
-      }
+      logger.info(msg); // should we leave this at info level ?
+
+      queue.channelClosed(new ChannelClosedException(msg));
```

Comment:
Nice catch Sudheesh.

Reflecting on this now, I think there is a much cleaner way to structure this than the current implementation. If we manage this at the connection level (new "outstanding handler => rpc id mapping" per connection), we can rely on Netty's RPC thread properties to have a trivial structure that doesn't need to be thread safe and is closed out on connection termination. Do one of you want to pick up that modification? I think that would make things much nicer.